pkg/controller/core/workload_controller_test.go

```diff
@@ -329,6 +355,42 @@ func TestReconcile(t *testing.T) {
 					State: kueue.CheckStateReady,
 				}).
 				Obj(),
+			wantWorkload: utiltesting.MakeWorkload("wl", "ns").
+				ReserveQuota(utiltesting.MakeAdmission("q1").Obj()).
+				Admitted(true).
+				AdmissionCheck(kueue.AdmissionCheckState{
+					Name:  "check",
+					State: kueue.CheckStateReady,
+				}).
+				Obj(),
+		},
+		"remove finalizer for finished workload": {
+			workload: utiltesting.MakeWorkload("unit-test", "ns").Finalizers(kueue.ResourceInUseFinalizerName).
+				Condition(metav1.Condition{
+					Type:   "Finished",
+					Status: "True",
+				}).
+				Delete().
+				Obj(),
+			wantWorkload: nil,
+		},
+		"don't remove finalizer for owned finished workload": {
+			workload: utiltesting.MakeWorkload("unit-test", "ns").Finalizers(kueue.ResourceInUseFinalizerName).
+				Condition(metav1.Condition{
+					Type:   "Finished",
+					Status: "True",
+				}).
+				OwnerReference(batchv1.SchemeGroupVersion.String(), "Job", "job", "test-uid", true, true).
+				Delete().
+				Obj(),
+			wantWorkload: utiltesting.MakeWorkload("unit-test", "ns").Finalizers(kueue.ResourceInUseFinalizerName).
+				Condition(metav1.Condition{
+					Type:   "Finished",
+					Status: "True",
+				}).
+				OwnerReference(batchv1.SchemeGroupVersion.String(), "Job", "job", "test-uid", true, true).
+				Delete().
```

Comment:
Comparing the `deletionTimestamp` would cause a flaky test?
So, can we verify if the workload has the `deletionTImestamp` in Line 427 instead of comparison?

Reply:
We can probably use `EquateApproxTime`.